src/stores/signaling.ts

```diff
@@ -0,0 +1,308 @@
+			const attendeeUsers = store.getters.participantsList(token) as Participant[]
+			const sessionIdsToUpdate: Record<string, string[]> = {}
+			let hasUnknownSessions = false
+
+			for (const participant of participants) {
+				const attendeeId = this.getSignalingSession(participant.sessionid)?.attendeeId
+					?? attendeeUsers.find(attendee => {
+						return participant.userid
+							? attendee.actorType !== ATTENDEE.ACTOR_TYPE.GUESTS && attendee.actorId === participant.userid
+							: attendee.actorType === ATTENDEE.ACTOR_TYPE.GUESTS && attendee.sessionIds.includes(participant.sessionid)
+					})?.attendeeId
+
+				this.addSignalingSession({
+					attendeeId,
+					token,
+					signalingSessionId: participant.sessionid,
+					sessionId: participant.roomsessionid,
+				})
+
+				if (!attendeeId) {
+					hasUnknownSessions = true
+					continue
+				}
+
+				const attendee = store.getters.getParticipant(token, attendeeId)
+
+				if (!sessionIdsToUpdate[attendeeId]) {
+					sessionIdsToUpdate[attendeeId] = [...attendee.sessionIds]
+				}
+				// Participant might join from several devices
+				sessionIdsToUpdate[attendeeId].push(participant.roomsessionid)
+			}
+
+			for (const [attendeeId, sessionIds] of Object.entries(sessionIdsToUpdate)) {
+				store.commit('updateParticipant', {
+					token,
+					attendeeId,
+					updatedData: { sessionIds },
+				})
+			}
+
+			return hasUnknownSessions
+		},
+
+		/**
+		 * Update participants left in store according to data from standalone signaling server
+		 *
+		 * @param signalingSessionIds disconnected signaling sessions;
+		 */
+		updateParticipantsLeftFromStandaloneSignaling(signalingSessionIds: string[]) {
+			for (const signalingSessionId of signalingSessionIds) {
+				const { token, attendeeId, sessionId } = this.getSignalingSession(signalingSessionId)
+				const attendee = store.getters.getParticipant(token, attendeeId)
+				if (!attendee) {
+					continue
+				}
+
+				const updatedData : { sessionIds: [], inCall?: number } = {
+					sessionIds: attendee.sessionIds.filter((id: string) => id !== sessionId)
+				}
+				if (attendee.sessionIds.length === 0) {
+					updatedData.inCall = PARTICIPANT.CALL_FLAG.DISCONNECTED
+				}
+				store.commit('updateParticipant', { token, attendeeId, updatedData })
+				this.deleteSignalingSession(signalingSessionId)
+			}
+		},
+
+		/**
+		 * Update participants changed in store according to data from standalone signaling server
+		 *
+		 * @param token the conversation token;
+		 * @param participants the changed participant objects;
+		 */
+		updateParticipantsChangedFromStandaloneSignaling(token: string, participants: StandaloneSignalingChangePayload) {
+			const guestNameStore = useGuestNameStore()
+			const attendeeUsersToUpdate: StandaloneUpdatePayload = {}
+
+			for (const participant of participants) {
+				const { token, attendeeId } = this.getSignalingSession(participant.sessionId)
+				if (!attendeeId) {
+					continue
+				}
```

Comment:
If we weren't able to find `attendeeId` in signaling store for some reason, at least we try to find it from participants store as a safeguard.

Reply:
With refactored method should be doable